llvm/test/CodeGen/SystemZ/builtin-setjmp-spills.ll

```diff
@@ -0,0 +1,255 @@
+; CHECK-NEXT:    mvhi 0(%r10), 1
+; CHECK-NEXT:    mvc 272(8,%r15), 312(%r15) # 8-byte Folded Reload
+; CHECK-NEXT:    mvc 264(8,%r15), 304(%r15) # 8-byte Folded Reload
+; CHECK-NEXT:    mvc 256(8,%r15), 296(%r15) # 8-byte Folded Reload
+; CHECK-NEXT:    mvc 248(8,%r15), 288(%r15) # 8-byte Folded Reload
+; CHECK-NEXT:    stg %r5, 240(%r15)
+; CHECK-NEXT:    stg %r14, 232(%r15)
+; CHECK-NEXT:    mvc 224(8,%r15), 280(%r15) # 8-byte Folded Reload
+; CHECK-NEXT:    stg %r9, 216(%r15)
+; CHECK-NEXT:    stg %r1, 208(%r15)
+; CHECK-NEXT:    stg %r2, 200(%r15)
+; CHECK-NEXT:    stg %r3, 192(%r15)
+; CHECK-NEXT:    lg %r9, 344(%r15) # 8-byte Folded Reload
+; CHECK-NEXT:    stg %r9, 184(%r15)
+; CHECK-NEXT:    stg %r4, 176(%r15)
+; CHECK-NEXT:    stg %r8, 168(%r15)
+; CHECK-NEXT:    stg %r7, 160(%r15)
+; CHECK-NEXT:    lgr %r2, %r10
+; CHECK-NEXT:    lgr %r3, %r11
+; CHECK-NEXT:    lgr %r4, %r12
+; CHECK-NEXT:    lgr %r5, %r13
+; CHECK-NEXT:    brasl %r14, foo@PLT
+; CHECK-NEXT:    l %r0, 0(%r11)
+; CHECK-NEXT:    a %r0, 0(%r10)
+; CHECK-NEXT:    a %r0, 0(%r12)
+; CHECK-NEXT:    a %r0, 0(%r13)
+; CHECK-NEXT:    a %r0, 0(%r6)
+; CHECK-NEXT:    a %r0, 0(%r7)
+; CHECK-NEXT:    a %r0, 0(%r8)
+; CHECK-NEXT:    lg %r1, 320(%r15) # 8-byte Folded Reload
+; CHECK-NEXT:    a %r0, 0(%r1)
+; CHECK-NEXT:    a %r0, 0(%r9)
+; CHECK-NEXT:    lg %r1, 328(%r15) # 8-byte Folded Reload
+; CHECK-NEXT:    a %r0, 0(%r1)
+; CHECK-NEXT:    lg %r1, 336(%r15) # 8-byte Folded Reload
+; CHECK-NEXT:    a %r0, 0(%r1)
+; CHECK-NEXT:    lg %r1, 352(%r15) # 8-byte Folded Reload
+; CHECK-NEXT:    a %r0, 0(%r1)
+; CHECK-NEXT:    lg %r1, 360(%r15) # 8-byte Folded Reload
+; CHECK-NEXT:    a %r0, 0(%r1)
+; CHECK-NEXT:    lg %r1, 280(%r15) # 8-byte Folded Reload
+; CHECK-NEXT:    a %r0, 0(%r1)
+; CHECK-NEXT:    lg %r1, 368(%r15) # 8-byte Folded Reload
+; CHECK-NEXT:    a %r0, 0(%r1)
+; CHECK-NEXT:    lg %r1, 376(%r15) # 8-byte Folded Reload
+; CHECK-NEXT:    a %r0, 0(%r1)
+; CHECK-NEXT:    lg %r1, 288(%r15) # 8-byte Folded Reload
+; CHECK-NEXT:    a %r0, 0(%r1)
+; CHECK-NEXT:    lg %r1, 296(%r15) # 8-byte Folded Reload
+; CHECK-NEXT:    a %r0, 0(%r1)
+; CHECK-NEXT:    lg %r1, 304(%r15) # 8-byte Folded Reload
+; CHECK-NEXT:    a %r0, 0(%r1)
+; CHECK-NEXT:    lg %r1, 312(%r15) # 8-byte Folded Reload
+; CHECK-NEXT:    a %r0, 0(%r1)
+; CHECK-NEXT:    lgfr %r2, %r0
+; CHECK-NEXT:    ld %f8, 440(%r15) # 8-byte Folded Reload
+; CHECK-NEXT:    ld %f9, 432(%r15) # 8-byte Folded Reload
+; CHECK-NEXT:    ld %f10, 424(%r15) # 8-byte Folded Reload
+; CHECK-NEXT:    ld %f11, 416(%r15) # 8-byte Folded Reload
+; CHECK-NEXT:    ld %f12, 408(%r15) # 8-byte Folded Reload
+; CHECK-NEXT:    ld %f13, 400(%r15) # 8-byte Folded Reload
+; CHECK-NEXT:    ld %f14, 392(%r15) # 8-byte Folded Reload
+; CHECK-NEXT:    ld %f15, 384(%r15) # 8-byte Folded Reload
+; CHECK-NEXT:    lmg %r6, %r15, 496(%r15)
+; CHECK-NEXT:    br %r14
+entry:
+  %0 = tail call i32 @llvm.eh.sjlj.setjmp(ptr nonnull @buf)
```

Comment:
Looking at this again, I'm a bit confused what this is supposed to test?  There are no registers live across the setjmp anyway ...

I'd have throught a good test would be to make a large number of virtual registers live *before* the setjmp, and then use all of them *after* the setjmp.  The test would then verify that the compiler doesn't attempt to keep any of these virtual registers alive in call-saved GPRs, but rather saves all of them to the stack.